.github/workflows/go-build-test.yml

```diff
@@ -15,30 +15,137 @@ on:
       - 'go.sum'
 
 jobs:
-  build-and-test:
+  build-and-test-multi-arch:
     runs-on: ubuntu-latest
+    name: Build on ${{ matrix.arch }}
+    # Run steps on a matrix of 2 arch.
+    strategy:
+      matrix:
+        arch:
+          - x64
+          - aarch64
     steps:
-      - uses: actions/checkout@v3
-
-      - name: Set up Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: '1.18'
-          check-latest: true
-
-      - name: Build UI
-        run: make ui/build
-
-      - name: Test
-        run: make go/test
-
-      - name: Build
-        run: make build
-
-      - name: Archive generatated artifacts
-        uses: actions/upload-artifact@v3
-        with:
-          name: parca-bin
-          if-no-files-found: error
-          path: |
-            bin
+    - uses: actions/checkout@v3
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: '1.18'
+        check-latest: true
+
+    - name: Build UI
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make ui/build
+
+    - name: Test
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make go/test
+
+    - name: Build
+      if: ${{ matrix.arch == 'x64' }}
+      run: |
+        make build
+
+    - name: Archive generatated artifacts
+      if: ${{ matrix.arch == 'x64' }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: parca-bin
+        if-no-files-found: error
+        path: |
+          bin
+
+    - name: 'Run ${{ matrix.arch }}'
```

Comment:
Even though this templating indicates this task is parametric and will work for a different architecture, it's actually not. There is a lot of hardcoded arm-specific configuration below. 

It would be awesome to have a task that would work for future architecture as well. If it's too much work, let's indicate that this is just for arm64.

Reply:
Added support for different architecture. When adding new architecture just add the name in strategy array and it should work. I added a `s390x` job on the same pipeline to test. I think it should work

Reply:
Ok so new error popped up on s390x arch test but I guess currently `ARCHS=('amd64' 'arm64' 'armv6' 'armv7' '386')` are the supported architecures right? cc @kakkoyun

Reply:
Yes, I think so. You can check the list of binaries we release in the latest release https://github.com/parca-dev/parca/releases/tag/v0.11.1

Reply:
As mentioned [here](https://github.com/parca-dev/parca/pull/1060#discussion_r886419828), I have added support for multi arch. But the tests on 32bit architecure is failing. Please check logs [here](https://github.com/parca-dev/parca/runs/6686157753). So currently removed adding another arch test other than arm64.  I guess that should be the scope of another issue/PR. WDYT @kakkoyun?

Reply:
We shouldn't actually support 32-bit. We should even stop building binaries for 32-bit arch. Let's exclude those architectures and I'll send a PR to stop releasing those binaries.

Reply:
Cool, then PR is ready for review then. PTAL whenever free

Reply:
FYI https://github.com/parca-dev/parca/pull/1115